### ISSUE TITLE
rename the execute function

### ISSUE THREAD

**Issue by Fabien-GELUS** [Substra/substra-tools#64]
In function.py we have 2 different execute functions, and it's really difficult when calling for instance `function.execute([`to know which one is used, we have to track the `function` object type.
I propose to rename one of them, like `execute_function` for the one in wrapper or `execute_cli` for the one running the cli

**Comment by ThibaultFy** [comment]
I agree that it is confusing. I propose to keep execute for the cli as it is the function exposed to the user, in order to keep a name as simple as possible, and to rename to execute_function for the wrapper.
